code/modules/atmospherics/machinery/datum_pipeline.dm

```diff
@@ -193,51 +193,43 @@
 	var/target_temperature
 	var/target_heat_capacity
 
-	if(isopenturf(target))
 
-		var/turf/open/modeled_location = target
-		target_temperature = modeled_location.GetTemperature()
-		target_heat_capacity = modeled_location.GetHeatCapacity()
+	var/turf/modeled_location = target
+	target_temperature = modeled_location.GetTemperature()
+	target_heat_capacity = modeled_location.GetHeatCapacity()
 
-		if(modeled_location.blocks_air)
+	if(modeled_location.blocks_air)
```

Comment:
why are you doing this check?  I know why I would do it, to prevent open turfs having temperature_expose called on them since it's really only a temporary thing in this case, but aren't these the same operation but phrased slightly differently? I don't understand

Reply:
Uh tbh it was there and I didn't want to touch it before a proper review.
How should I change it?

Reply:
It's literally the same math, so cull the duplicate code, and only call temperature_expose() on the turf that blocks air. That should do it

Reply:
done